utils.go

```diff
@@ -87,12 +87,11 @@ func getRootlessRuntimeDir(rootlessUID int) (string, error) {
 	if err == nil {
 		return runtimeDir, nil
 	}
-	tmpDir := fmt.Sprintf("/run/user/%d", rootlessUID)
+	tmpDir := fmt.Sprintf("%s/storage-user-%d", os.TempDir(), rootlessUID)
 	st, err := system.Stat(tmpDir)
```

Comment:
`Stat` follows symlinks, which makes the ID check below not safe with predictable file names.

Reply:
I see no difference between `/run` and `/tmp` prefix. This patch adds no additional problems.

Reply:
The big difference is that unprivileged users are not allowed to create new directories in `/run/user`.

(If you don’t understand the function and the minute details of the various directories, consider whether it is reasonable to be strongly demanding specific solutions to be chosen.)

Reply:
@mtrmac If we switch the os.Stat to os.Lstat and verify that the path is a directory, it should be save, correct?

Reply:
It’s certainly _more_ safe. But if we are considering systems without `/run/user`, the `/run/user` path could be an attacker-created symlink. Ugh.

Reply:
Actually, `/run` is root-writable-only (if it exists at all), so this should be safe then. But yes, this **absolutely must** be `Lstat` ⚠️